contrib/test.sh

```diff
@@ -2,6 +2,14 @@
 
 FEATURES="base64 bitcoinconsensus use-serde rand"
 
+# Old cargo format is incompatible with a newer ones, so we need to clean this up
+rm -f Cargo.lock
```

Comment:
Since this PR tells devs to use this script locally too I'm not sure if we should just delete the lockfile. Maybe print a warning instead if using an old rust version? The user will notice. Or at least gate it with a flag.

Reply:
I will add NB about this to the CONTRIBUTING

Reply:
Did that below